pkg/controller/infrastructureconfig/sync_vips.go

```diff
@@ -17,38 +17,39 @@ type apiAndIngressVipsSynchronizer struct{}
 
 // VipsSynchronize synchronizes new API & Ingress VIPs with old fields.
 // It returns if the status was updated and the updated infrastructure object.
+//
 //nolint:staticcheck
 func (*apiAndIngressVipsSynchronizer) VipsSynchronize(infraConfig *configv1.Infrastructure) *configv1.Infrastructure {
 	var apiVIPs, ingressVIPs *[]string // new fields
 	var apiVIP, ingressVIP *string     // old/deprecated fields
 
 	updatedInfraConfig := infraConfig.DeepCopy()
-	switch updatedInfraConfig.Status.Platform {
-	case configv1.BareMetalPlatformType:
+	switch {
+	case updatedInfraConfig.Status.Platform == configv1.BareMetalPlatformType:
 		apiVIPs = &updatedInfraConfig.Status.PlatformStatus.BareMetal.APIServerInternalIPs
 		apiVIP = &updatedInfraConfig.Status.PlatformStatus.BareMetal.APIServerInternalIP
 		ingressVIPs = &updatedInfraConfig.Status.PlatformStatus.BareMetal.IngressIPs
 		ingressVIP = &updatedInfraConfig.Status.PlatformStatus.BareMetal.IngressIP
 
-	case configv1.VSpherePlatformType:
+	case updatedInfraConfig.Status.Platform == configv1.VSpherePlatformType && updatedInfraConfig.Status.PlatformStatus.VSphere != nil: //VSphere UPI doesn't populate VSphere field
```

Comment:
nit: space after the double comment line. `// VSphere`

Reply:
:+1 updated it